README.md

```diff
@@ -42,8 +42,45 @@ data:
 
 ## Usage
 
+### Supported Backends
+As of today argocd-vault-plugin supports HashiCorp Vault and IBM Secret Manger as backends.
+
+#### HashiCorp Vault
+We support AppRole and Github Auth Method for getting secrets from Vault.
+
+##### AppRole Authentication
+For AppRole Authentication, these are the required environment variables:
+```
+AVP_VAULT_ADDR: Your HashiCorp Vault Address
+AVP_TYPE: vault
+AVP_AUTH_TYPE: approle
+AVP_ROLE_ID: Your AppRole Role ID
+AVP_SECRET_ID: Your AppRole Secret ID
+```
+
+##### Github Authentication
+For Github Authentication, these are the required environment variables:
+```
+AVP_VAULT_ADDR: Your HashiCorp Vault Address
+AVP_TYPE: vault
+AVP_AUTH_TYPE: github
+AVP_GITHUB_TOKEN: Your Github Personal Access Token
+```
+
+#### IBM Secret Manager
+For IBM Secret Manager we only support using IAM authentication at this time.
+
+##### IAM Authentication
+For IAM Authentication, these are the required environment variables:
+```
+AVP_VAULT_ADDR: Your IBM Secret Manager Endpoint
+AVP_TYPE: secretmanager
+AVP_AUTH_TYPE: iam
+AVP_IBM_API_KEY: Your IBM Cloud API Key
+```
+
 ### Configuration
-The plugin requires some configuration to connect to Vault. The parameters are:
+The full list of support environment variables are:
```

Comment:
These aren't the env variables, i.e, they won't work w/o putting the `AVP_` prefix. That won't be clear if you keep this rephrasing